test/remediation.test.ts

```diff
@@ -0,0 +1,68 @@
+import { test } from 'tap';
+import fs = require('fs');
+
+import plugin = require('../lib');
+import { chdirWorkspaces } from './test-utils';
+const testUtils = require('./test-utils') as any;
+
+test('remediation with constraints.txt', async (t) => {
+  chdirWorkspaces('pip-app');
+  t.teardown(testUtils.activateVirtualenv('pip-app'));
+  const upgrades = {
+    'Django@1.6.1': { upgradeTo: 'Django@2.0.1' },
+    'transitive@1.0.0': { upgradeTo: 'transitive@1.1.1' },
+  };
+  const manifests = {
+    'requirements.txt': fs.readFileSync('requirements.txt', 'utf8'),
+    'constraints.txt': '',
```

Comment:
what is `constraints.txt`?

Reply:
`constraints` files are part of pips processing , and will be part of future remediation options. https://pip.pypa.io/en/stable/user_guide/#constraints-files

Reply:
do we support this now as well when scanning? Do we have tests to show what we support with it?

Reply:
We do de-facto support them for CLI scans (because the user is free to install the deps whichever way they like).
As for the SCM support: what's the status on "project root"? Do we have a "proper" way of handling multi-file manifests?